compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -265,14 +265,45 @@ public void visit(BLangPackage pkgNode) {
                 sortedListOfNodes.add(topLevelNode);
             }
         });
-        sortedListOfNodes.forEach(topLevelNode -> analyzeNode((BLangNode) topLevelNode, env));
+
+        sortedListOfNodes.forEach(topLevelNode -> {
+            if (isModuleInitFunction((BLangNode) topLevelNode)) {
+                analyzeModuleInitFunc((BLangFunction) topLevelNode);
+                checkForUninitializedGlobalVar(pkgNode.globalVars);
+            } else {
+                analyzeNode((BLangNode) topLevelNode, env);
+            }
+        });
         pkgNode.getTestablePkgs().forEach(testablePackage -> visit((BLangPackage) testablePackage));
         this.globalVariableRefAnalyzer.analyzeAndReOrder(pkgNode, this.globalNodeDependsOn);
         this.globalVariableRefAnalyzer.populateFunctionDependencies(this.functionToDependency);
         checkUnusedImports(pkgNode.imports);
         pkgNode.completedPhases.add(CompilerPhase.DATAFLOW_ANALYZE);
     }
 
+    private boolean isModuleInitFunction(BLangNode node) {
+        if (node.getKind() == NodeKind.FUNCTION &&
+                Names.USER_DEFINED_INIT_SUFFIX.value.equals(((BLangFunction) node).name.value)) {
+            return true;
+        }
+        return false;
+    }
+
+    private void analyzeModuleInitFunc(BLangFunction funcNode) {
```

Comment:
Why do we do a separate analysis for the module init?

Anyway, we're missing annotation validations from `visit(BLangFunction funcNode)`.

For example,

```ballerina
annotation map<int> Foo on function;

int i; // uninitialized variable 'i'

@Foo {
    i // NO error, expected `variable 'i' is not initialized`
}
function __init() {
}

@Foo {
    i // variable 'i' is not initialized
}
public function main() {
   int j = i; // variable 'i' is not initialized
}
```

Reply:
In current implementation Analyse function call the analyzeBranch method and it does not modify `uninitializedVars`. Because module-level variables cannot be initialized in function (same as __init() in the object).  But in this scenario, if we see some variable is initialized we need to update that map. Annotations are validated.